test/sqllogictest/tpch.slt

```diff
@@ -161,7 +161,7 @@ ORDER BY
 %0 =
 | Get materialize.public.lineitem (u21)
 | Filter (#10 <= 1998-12-01)
-| Reduce group=(#8, #9) sum(#4) sum(#5) sum((#5 * (100dec - #6))) sum(((#5 * (100dec - #6)) * (100dec + #7))) countall(null) countall(null) sum(#6) countall(null) countall(true)
+| Reduce group=(#8, #9) sum(#4) sum(#5) sum((#5 * (100dec - #6))) sum(((#5 * (100dec - #6)) * (100dec + #7))) count(#4) count(#5) sum(#6) count(#6) count(true)
```

Comment:
This might be an example of that optimization, where previously we were able to remove the dependence on `#4` and `#5` but now do not. We obvs still need them (in earlier aggregations) but if we ditched them we could deduplicate the aggregations more easily (I guess we don't do that, though!).

Reply:
Ooo, great catch! I didn't look at this plan movement very in depth because I figured it would be just like the chbench movement, but evidently not.

The optimization is restored with the approach you suggest and seems to be working great now!